fabric-server-languages-v1/src/main/java/net/fabricmc/fabric/mixin/server/languages/v1/MixinLanguage.java

```diff
@@ -0,0 +1,78 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.server.languages.v1;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.gson.JsonParseException;
+import org.apache.logging.log4j.Logger;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+
+import net.minecraft.util.Language;
+
+import net.fabricmc.loader.api.FabricLoader;
+
+import java.io.InputStream;
+import java.io.IOException;
+import java.util.function.BiConsumer;
+import java.util.HashMap;
+
+@Mixin(Language.class)
+public class MixinLanguage {
+	@Shadow
+	@Final
+	private static Logger LOGGER;
+	@Shadow
+	@Final
+	private static String DEFAULT_LANGUAGE;
+
+	@Shadow
+	private static void load(InputStream inputStream, BiConsumer<String, String> entryConsumer) {
+	}
+
+	private static void loadModLangFile(String modId, BiConsumer<String, String> biConsumer) {
+		String path = "/assets/" + modId + "/lang/" + DEFAULT_LANGUAGE + ".json";
+
+		try (InputStream inputStream = Language.class.getResourceAsStream(path)) {
+			if (inputStream == null) return;
+			load(inputStream, biConsumer);
+		} catch (JsonParseException | IOException e) {
+			LOGGER.error("Couldn't read strings from " + path, e);
+		}
+	}
+
+	@Inject(method = "create", locals = LocalCapture.CAPTURE_FAILSOFT, at = @At(value = "INVOKE", remap = false, target = "Lcom/google/common/collect/ImmutableMap$Builder;build()Lcom/google/common/collect/ImmutableMap;"))
+	private static void create(CallbackInfoReturnable<Language> cir, ImmutableMap.Builder<String, String> builder) {
+		/*  We must ensure that the keys are de-duplicated because we can't catch the error that might otherwise
+		 *  occur when the injected function calls build() on the ImmutableMap builder. So we use our own hash map and
+		 *  exclude "minecraft", as the injected function has already loaded those keys at this point.
+		 */
+		HashMap<String, String> translations = new HashMap<>();
+
+		FabricLoader.getInstance().getAllMods().stream().map(modContainer -> modContainer.getMetadata().getId())
+				.filter(id -> !id.equals("minecraft")).forEach(id -> {
+					loadModLangFile(id, translations::put);
```

Comment:
Here, we want to have a strategy for handling duplicate translations, given mods are iterated in arbitrary order without dependency specification, etc.

Reply:
Is this something that could be done if we follow @LambdAurora's suggestion and integrate with ModResourcePacks? Not sure what data we have available to resolve conflicts, or iterate in a manner that respects some priority?

Reply:
I've also realized that I'm making a bad assumption about mods not overwriting minecraft's keys, which are in the map builder already. But that's just part of this broader problem: figuring out who gets to overwrite whose keys when we don't have resource pack loading logic.

Reply:
I sketched out a strategy for handling duplicates using mod metadata in the latest commit.